src/editor-service/editor-server.ts

```diff
@@ -0,0 +1,167 @@
+ * some modification of method calls for the JSON format. See the Request
+ * type for more information.
+ */
+export interface RequestWrapper {
+  id: number;
+  value: Request;
+}
+export interface ResponseWrapper {
+  id: number;
+  value: SettledValue;
+}
+export type SettledValue = Resolution | Rejection;
+export interface Resolution {
+  kind: 'resolution';
+  resolution: any;
+}
+export interface Rejection {
+  kind: 'rejection';
+  rejection: string;
+}
+
+
+export type Request = InitRequest | FileChangedRequest | GetWarningsRequest |
+    GetDocumentationRequest | GetDefinitionRequest |
+    GetTypeaheadCompletionsRequest | ClearCachesRequest;
+export interface InitRequest {
+  kind: 'init';
+  basedir: string;
+}
+export interface FileChangedRequest {
+  kind: 'fileChanged';
+  localPath: string;
+  contents?: string;
+}
+export interface GetWarningsRequest {
+  kind: 'getWarningsFor';
+  localPath: string;
+}
+export interface GetDocumentationRequest {
+  kind: 'getDocumentationFor';
+  localPath: string;
+  position: SourcePosition;
+}
+export interface GetDefinitionRequest {
+  kind: 'getDefinitionFor';
+  localPath: string;
+  position: SourcePosition;
+}
+export interface GetTypeaheadCompletionsRequest {
+  kind: 'getTypeaheadCompletionsFor';
+  localPath: string;
+  position: SourcePosition;
+}
+export interface ClearCachesRequest { kind: 'clearCaches'; }
+
+
+/**
+ * Runs out of process and handles decoded Requests.
+ */
+class EditorServer {
+  private _localEditorService: LocalEditorService;
+  constructor(basedir: string) {
+    this._localEditorService = new LocalEditorService({
+      urlLoader: new FSUrlLoader(basedir),
+      urlResolver: new PackageUrlResolver()
+    });
+  }
+
+  async handleMessage(message: Request): Promise<any> {
+    switch (message.kind) {
+      case 'getWarningsFor':
+        return this._localEditorService.getWarningsFor(message.localPath);
+      case 'fileChanged':
+        await this._localEditorService.fileChanged(
+            message.localPath, message.contents);
+        return;
+      case 'init':
+        throw new Error('Already initialized!');
+      case 'getDefinitionFor':
+        return this._localEditorService.getDefinitionFor(
+            message.localPath, message.position);
+      case 'getDocumentationFor':
+        return this._localEditorService.getDocumentationFor(
+            message.localPath, message.position);
+      case 'getTypeaheadCompletionsFor':
+        return this._localEditorService.getTypeaheadCompletionsFor(
+            message.localPath, message.position);
+      case 'clearCaches':
+        return this._localEditorService.clearCaches();
+      default:
+        // This assignment makes it a type error if we don't handle all possible
+        // values of `message.kind`.
+        const never: never = message;
+        throw new Error(`Got unknown kind of message: ${util.inspect(never)}`);
+    }
+  }
+}
+
+if (!module.parent) {
```

Comment:
It's some sort of reference to the importing node library. editor-server now doesn't export anything and should only ever be used as a binary. Removed.